staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -641,7 +645,10 @@ func (v *specStandardValidatorV3) validate(schema *apiextensions.JSONSchemaProps
 	}
 
 	if schema.Type == "null" {
-		allErrs = append(allErrs, field.Forbidden(fldPath.Child("type"), "type cannot be set to null"))
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("type"), "type cannot be set to null, use nullable as an alternative"))
+	}
+	if schema.Nullable && schema.Type != "object" && schema.Type != "array" {
```

Comment:
what do we generate for pointers to scalars?

Reply:
there we usually have omitempty. We could add that as well here, but I have never seen such a use-case. Have you?

Reply:
I have now generalized nullable to any type (also without where it has no effect, but is perfectly valid). Please take a look.